.clang-format

```diff
@@ -66,15 +66,52 @@ ForEachMacros:
   - Q_FOREACH
   - BOOST_FOREACH
 IncludeBlocks:   Regroup
+# Include Groups (by priority):
+#   1. C system headers:     `#include <*.h>`
+#
+#   2. C++ system headers:   `#include <*>`
+#
+#   3. PyTorch headers:
+#     a) API headers:        `#include <torch/*.h>`
+#     b) ATen headers:       `#include <ATen/*.h>`
+#     c) c10 headers:        `#include <c10/*.h>`
+#
+#     Note that PyTorch headers should be included with angle-brackets so as to
+#     keep consistency with PyTorch source code. This should also prevent
+#     headers to be included twice because of different paths.
+#
+#     Google C++ Style Guide dictates (3) and (4) should belong to the same
+#     group. However, this breaks PyTorch/XLA build. The problem is that both
+#     PyTorch and abseil-cpp (included by OpenXLA) define (among many) the
+#     `LOG()` macro. This leads to problems, such as OpenXLA functions calling
+#     PyTorch `LOG()` macro due to include order.
+#
+#     In order to avoid such a problem, we make PyTorch into its own group,
+#     just after the system libraries. This works because including OpenXLA
+#     headers will redefine PyTorch `LOG()` macro, solving the problem above.
+#     This, however might fail one day, if PyTorch starts calling `LOG()` macro
+#     in one of the headers we include in PyTorch/XLA.
```

Comment:
I had to move PyTorch headers to their own group, right after the C++ system libraries. Otherwise, we would get a build error where [some OpenXLA files][3] would call the `LOG()` macro [defined by PyTorch][1], rather than the one [defined by `abseil-cpp`][2]. See the comment above for more details.

[1]: https://github.com/pytorch/pytorch/blob/21fec65781bebe867faf209f89bb687ffd236ca4/c10/util/logging_is_not_google_glog.h#L92-L94
[2]: https://github.com/abseil/abseil-cpp/blob/987c57f325f7fa8472fa84e1f885f7534d391b0d/absl/log/log.h#L218
[3]: https://github.com/openxla/xla/blob/9a9aa0e11e4fcda8d6a9c3267dca6776ddbdb0ca/xla/pjrt/distributed/client.h#L73-L83